plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/DeployPreferencesDialog.java

```diff
@@ -0,0 +1,116 @@
+package com.google.cloud.tools.eclipse.appengine.deploy.ui;
+
+import org.eclipse.core.databinding.ValidationStatusProvider;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.jface.databinding.dialog.TitleAreaDialogSupport;
+import org.eclipse.jface.databinding.dialog.ValidationMessageProvider;
+import org.eclipse.jface.dialogs.Dialog;
+import org.eclipse.jface.dialogs.IDialogConstants;
+import org.eclipse.jface.dialogs.IMessageProvider;
+import org.eclipse.jface.dialogs.TitleAreaDialog;
+import org.eclipse.jface.layout.GridDataFactory;
+import org.eclipse.swt.graphics.Image;
+import org.eclipse.swt.layout.GridLayout;
+import org.eclipse.swt.widgets.Button;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Control;
+import org.eclipse.swt.widgets.Layout;
+import org.eclipse.swt.widgets.Shell;
+import org.eclipse.ui.forms.events.ExpansionEvent;
+import org.eclipse.ui.forms.widgets.ExpandableComposite;
+
+import com.google.cloud.tools.eclipse.appengine.deploy.ui.DeployPreferencesPanel.SectionExpansionHandler;
+import com.google.cloud.tools.eclipse.appengine.ui.AppEngineImages;
+import com.google.common.base.Preconditions;
+
+public class DeployPreferencesDialog extends TitleAreaDialog {
+
+  // if the image is smaller (e.g. 32x32, it will break the layout of the TitleAreaDialog)
+  // seems like an Eclipse/JFace bug
+  private Image titleImage = AppEngineImages.appEngine(64).createImage();
+
+  private DeployPreferencesPanel content;
+  private IProject project;
+
+  public DeployPreferencesDialog(Shell parentShell, IProject project) {
+    super(parentShell);
+
+    Preconditions.checkNotNull(project, "project is null");
+    this.project = project;
+  }
+
+  @Override
+  protected Control createContents(Composite parent) {
+    Control contents = super.createContents(parent);
+
+    getShell().setText(Messages.getString("deploy.preferences.dialog.title"));
+    setTitle(Messages.getString("deploy.preferences.dialog.title.withProject", project.getName()));
+
+    if (titleImage != null) {
+      setTitleImage(titleImage);
+    }
+
+    // TitleAreaDialogSupport does not validate initially, let's trigger validation this way
+    content.getDataBindingContext().updateTargets();
+
+    return contents;
+  }
+  
+  @Override
+  protected Control createDialogArea(final Composite parent) {
+    Composite dialogArea = (Composite) super.createDialogArea(parent);
+    setMargin(dialogArea);
```

Comment:
So `setMargin()` doesn't produce the right results as you can see in the following image: note the separator is now indented.
<img width="94" alt="screen shot 2016-09-12 at 9 51 25 pm" src="https://cloud.githubusercontent.com/assets/202851/18459691/b2ffe72a-7939-11e6-8a96-4388756043f6.PNG">
It turns out `TitleAreaDialog` overrides `Dialog#createDialogArea()` and creates the content area `Composite` with 0 margins so that the separator (shown above) goes to the edge.  What it should have done is create and return a child composite with the same margins.

I'll push a change that DTRT here, following practices in JDT to create a child composite and applying the same margins and spacing as was applied in `Dialog#createDialogArea()`.